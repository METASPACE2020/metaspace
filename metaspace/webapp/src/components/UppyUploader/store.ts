// based on https://uppy.io/docs/stores/#Implementing-Stores

import { ref } from 'vue'
import { State } from '@uppy/core'

type Listener = (prevState: State, nextState: State, patch: object) => void

export default () => {
  const state = ref<State | any>({})
  const listeners = new Set<Listener>()

  return {
    getState: () => state.value,
    setState: (patch: object) => {
      const prevState = state.value
      const nextState = { ...prevState, ...patch }

      state.value = nextState

      listeners.forEach((listener) => {
        listener(prevState, nextState, patch)
      })
    },
    subscribe: (listener: Listener) => {
      listeners.add(listener)
      return () => listeners.delete(listener)
    },
  }
}
