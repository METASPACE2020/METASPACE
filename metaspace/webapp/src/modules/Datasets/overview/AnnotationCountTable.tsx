import { DatasetAnnotationCount } from '../../../api/dataset'
import { defineComponent } from '@vue/composition-api'
import { encodeParams } from '../../Filters'

interface AnnotationCountTableData{
  name: string
  [key: string]: any // dynamic keys according to fdrs
}

interface AnnotationCountTableProps {
  id: string
  sumRowLabel: string
  btnLabel: string
  title: string
  data: DatasetAnnotationCount[]
  header: string[]
  headerTitleSuffix: string
}

export const AnnotationCountTable = defineComponent<AnnotationCountTableProps>({
  name: 'AnnotationCountTable',
  props: {
    id: { type: String, default: '' },
    title: { type: String, default: '# of annotations with FDR% <=' },
    sumRowLabel: { type: String, default: 'Total Annotations' },
    btnLabel: { type: String, default: 'Browse annotations' },
    data: { type: Array, default: () => [] },
    header: { type: Array, default: () => [5, 10, 20, 50] },
    headerTitleSuffix: { type: String, default: '%' },
  },
  setup(props, ctx) {
    const annotationsLink = (datasetId: string, database?: string, fdrLevel?: number) => ({
      name: 'dataset-annotations',
      params: { datasetId },
      query: encodeParams({
        database,
        fdrLevel,
      }),
    })

    const mountTableData = (data: DatasetAnnotationCount[], header: string[]) : AnnotationCountTableData[] => {
      if (!data) { return [] }

      return data.map((item: DatasetAnnotationCount) => {
        const obj = {} as AnnotationCountTableData
        obj.id = item?.databaseId
        obj.name = `${item?.dbName}${item?.dbVersion ? `-${item?.dbVersion}` : ''}`
        header.forEach((headerEl: string) => {
          obj[headerEl] = Array.isArray(item?.counts)
            ? (item.counts.find((count: any) => count.level === headerEl) || {}).n : '-'
        })
        return obj
      })
    }

    const browseAnnotations = () => {
      const { id } = props
      ctx.root.$router.push(annotationsLink(id))
    }

    const formatTitle = (col: number|string) => {
      const { headerTitleSuffix } = props
      return `${col === 'name' ? 'Database' : col}${col !== 'name' ? headerTitleSuffix : ''}`
    }

    const formatCell = (row: any, column: any, cellValue: any, rowIndex: number, colIndex: number) => {
      const { id } = props
      if (colIndex === 0) { return cellValue }

      return (
        <router-link to={annotationsLink(id?.toString(), row?.id,
          parseInt(column?.property, 10) / 100)}>
          {cellValue}
        </router-link>
      )
    }

    return () => {
      const { data, header, sumRowLabel, btnLabel, title } = props
      const tableData = mountTableData(data, header)

      return (
        <div class="relative text-center">
          <h4>{title}</h4>
          <el-table
            data={tableData}
            show-summary={true}
            sum-text={sumRowLabel}
            style="width: 100%; margin-top: 20px">
            {
              ['name'].concat(header).map((col, colIndex) => {
                return <el-table-column
                  sortable={true}
                  key={colIndex}
                  prop={col.toString()}
                  formatter={(row: any, column: any, cellValue: any, index: number) =>
                    formatCell(row, column, cellValue, index, colIndex)}
                  label={formatTitle(col)} />
              })
            }
          </el-table>
          <div class="text-right mt-2">
            <el-button onClick={browseAnnotations}>{btnLabel}</el-button>
          </div>
        </div>
      )
    }
  },
})
