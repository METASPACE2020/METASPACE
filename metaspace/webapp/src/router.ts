/* eslint-disable vue/max-len */
import Vue, { AsyncComponent } from 'vue'
import VueRouter, { RawLocation } from 'vue-router'
import AboutPage from './modules/App/AboutPage.vue'
import DatasetsPage from './modules/Datasets/DatasetsPage.vue'
import { DialogPage, ResetPasswordPage } from './modules/Account'
import { redirectAfterSignIn } from './modules/Account/signInReturnUrl'
import { updateDBParam } from './modules/Filters/url'
import NotFoundPage from './modules/App/NotFoundPage.vue'

Vue.use(VueRouter)

const asyncPagesFreelyTyped = {
  AnnotationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "AnnotationsPage" */ './modules/Annotations/AnnotationsPage.vue'),
  DatasetSummary: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetSummary" */ './modules/Datasets/summary/DatasetSummary.vue'),
  MetadataEditPage: () => import(/* webpackPrefetch: true, webpackChunkName: "MetadataEditPage" */ './modules/MetadataEditor/MetadataEditPage.vue'),
  ImageAlignmentPage: () => import(/* webpackPrefetch: true, webpackChunkName: "ImageAlignmentPage" */ './modules/ImageAlignment/ImageAlignmentPage.vue'),
  UploadPage: () => import(/* webpackPrefetch: true, webpackChunkName: "UploadPage" */ './modules/MetadataEditor/UploadPage.vue'),
  DesignPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DesignPage" */ './design/DesignPage.vue'),

  // These pages are relatively small as they don't have any big 3rd party dependencies, so pack them together
  DatasetTable: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Datasets/list/DatasetTable.vue'),
  HelpPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/HelpPage.vue'),
  EditUserPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/UserProfile/EditUserPage.vue'),
  CreateGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/GroupProfile/CreateGroupPage.vue'),
  ProjectsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Project/ProjectsListPage.vue'),
  SystemHealthPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Admin/SystemHealthPage.vue'),
  GroupsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Admin/GroupsListPage.vue'),
  PrivacyPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/PrivacyPage.vue'),
  TermsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/TermsPage.vue'),
  PublicationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/PublicationsPage.vue'),

  // These pages use sanitizeHtml, which is big
  ViewGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ './modules/GroupProfile/ViewGroupPage.vue'),
  ViewProjectPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ './modules/Project/ViewProjectPage.vue'),
}
const asyncPages = asyncPagesFreelyTyped as Record<keyof typeof asyncPagesFreelyTyped, AsyncComponent>

const convertLegacyUrls = () => {
  const { pathname, hash, search } = window.location
  if (pathname === '/' && hash && hash.startsWith('#/')) {
    history.replaceState(undefined, undefined as any, hash.slice(1))
  }
  if (pathname === '/annotations') {
    const updatedQueryString = updateDBParam(search.slice(1))
    if (updatedQueryString !== null) {
      history.replaceState(undefined, undefined as any, `${pathname}?${updatedQueryString}`)
    }
  }
}
convertLegacyUrls()

const router = new VueRouter({
  mode: 'history',
  routes: [
    { path: '/(about)?', component: AboutPage, meta: { footer: true, headerClass: 'bg-primary' } },
    { path: '/annotations', component: asyncPages.AnnotationsPage },
    {
      path: '/datasets',
      component: DatasetsPage,
      children: [
        { path: '', component: asyncPages.DatasetTable },
        { path: 'summary', component: asyncPages.DatasetSummary },
      ],
    },
    { path: '/datasets/edit/:dataset_id', name: 'edit-metadata', component: asyncPages.MetadataEditPage },
    { path: '/datasets/:dataset_id/add-optical-image', name: 'add-optical-image', component: asyncPages.ImageAlignmentPage },
    { path: '/upload', component: asyncPages.UploadPage },
    { path: '/help', component: asyncPages.HelpPage, meta: { footer: true } },
    { path: '/user/me', component: asyncPages.EditUserPage },

    { path: '/admin/health', component: asyncPages.SystemHealthPage },
    { path: '/admin/groups', component: asyncPages.GroupsListPage },

    { path: '/account/sign-in', component: DialogPage, props: { dialog: 'signIn' } },
    { path: '/account/sign-in-success', redirect: redirectAfterSignIn },
    { path: '/account/create-account', component: DialogPage, props: { dialog: 'createAccount' } },
    { path: '/account/forgot-password', component: DialogPage, props: { dialog: 'forgotPassword' } },
    { path: '/account/reset-password', component: ResetPasswordPage },

    { path: '/group/create', component: asyncPages.CreateGroupPage },
    { path: '/group/:groupIdOrSlug', name: 'group', component: asyncPages.ViewGroupPage },

    { path: '/project/:projectIdOrSlug', name: 'project', component: asyncPages.ViewProjectPage },
    { // Legacy URL sent in "request access" emails up until Feb 2019
      path: '/project/:projectIdOrSlug/manage',
      redirect: { path: '/project/:projectIdOrSlug', query: { tab: 'members' } } as RawLocation,
    },
    { path: '/projects', component: asyncPages.ProjectsListPage },

    { path: '/terms', component: asyncPages.TermsPage, meta: { footer: true } },
    { path: '/privacy', component: asyncPages.PrivacyPage, meta: { footer: true } },
    { path: '/publications', component: asyncPages.PublicationsPage, meta: { footer: true } },

    { path: '/design', component: asyncPages.DesignPage, meta: { footer: true, flex: true } },

    { path: '*', component: NotFoundPage, meta: { footer: true, flex: true } },
  ],
  scrollBehavior(to, from, savedPosition) {
    if (to.hash) {
      return { selector: to.hash, offset: { x: 0, y: 64 } } // offset header height
    }
    if (savedPosition) {
      return savedPosition
    }
    if (to.path !== from.path) {
      return { x: 0, y: 0 }
    }
  },
})

const { href } = router.resolve({ name: 'project', params: { projectIdOrSlug: 'REMOVE' } }, undefined, true)
export const PROJECT_URL_PREFIX = location.origin + href.replace('REMOVE', '')

export default router
