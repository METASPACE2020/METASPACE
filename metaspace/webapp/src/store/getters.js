import {mzFilterPrecision} from '../util';
import {decodeParams, decodeSettings, getLevel} from '../modules/Filters';
import config from '../config';

/** For filters where empty string is a valid client-side value that must be converted to empty string for the API */
const noneToEmptyString = s => s === 'none' ? '' : s;

export default {
  filterLevel(state) {
    return getLevel(state.route.path);
  },

  filter(state) {
    return decodeParams(state.route, state.filterLists);
  },

  settings(state) {
    return decodeSettings(state.route);
  },

  ftsQuery(state, getters) {
    return getters.filter.simpleQuery;
  },

  gqlAnnotationFilter(state, getters) {
    const filter = getters.filter;
    const colocalizationAlgo = getters.settings.annotationView.colocalizationAlgo;

    const f = {
      database: filter.database,
      compoundQuery: filter.compoundName,
      chemMod: noneToEmptyString(filter.chemMod),
      neutralLoss: noneToEmptyString(filter.neutralLoss),
      adduct: filter.adduct,
      fdrLevel: filter.fdrLevel,
      colocalizedWith: filter.colocalizedWith,
      // Only include colocalizationAlgo if there is another filter that uses it. Otherwise the annotations list
      // refreshes unnecessarily when changing algorithm.
      colocalizationAlgo: filter.colocalizedWith || filter.colocalizationSamples ? colocalizationAlgo : null,
      colocalizationSamples: filter.colocalizationSamples,
      offSample: filter.offSample == null ? undefined : !!filter.offSample,
    };

    if (!config.features.neutral_losses && !config.features.advanced_ds_config) {
      f.hasNeutralLoss = false;
    }

    if (!config.features.chem_mods && !config.features.advanced_ds_config) {
      f.hasChemMod = false;
    }

    if (!config.features.all_adducts) {
      f.hasHiddenAdduct = false;
    }

    if (filter.minMSM)
      f.msmScoreFilter = {min: filter.minMSM, max: 1.0};

    if (filter.mz) {
      const mz = parseFloat(filter.mz),
            deltamz = parseFloat(mzFilterPrecision(mz));
      f.mzFilter = {
        min: mz - deltamz,
        max: mz + deltamz
      };
    }

    return f;
  },

  gqlDatasetFilter(state, getters) {
    const filter = getters.filter;
    const {group, project, submitter, datasetIds, polarity,
           organism, organismPart, condition, growthConditions,
           ionisationSource, analyzerType, maldiMatrix, metadataType,
           compoundName} = filter;
    const level = getters.filterLevel;
    const hasAnnotationMatching = level === 'dataset' && compoundName ? { compoundQuery: compoundName } : undefined;
    return {
      group: group,
      project: project,
      submitter: submitter,

      // temporary workaround because of array-related bugs in apollo-client
      ids: datasetIds ? datasetIds.join("|") : null,

      organism,
      organismPart,
      condition,
      growthConditions,
      ionisationSource,
      maldiMatrix,
      analyzerType,
      polarity: polarity ? polarity.toUpperCase() : null,
      metadataType,
      hasAnnotationMatching,
    }
  },

  gqlColocalizationFilter(state, getters) {
    const {datasetIds, colocalizedWith, database, fdrLevel} = getters.filter;
    const colocalizationAlgo = getters.settings.annotationView.colocalizationAlgo;
    if (datasetIds && !datasetIds.includes('|') && colocalizedWith != null && database != null && fdrLevel != null) {
      return {colocalizedWith, colocalizationAlgo, database, fdrLevel};
    } else {
      return null;
    }
  },
}
