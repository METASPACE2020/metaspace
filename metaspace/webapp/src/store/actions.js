import { fetchOptionListsQuery } from '../api/metadata'
import { decodeParams } from '../modules/Filters'
import { computed } from 'vue'
import store from '../store/index'

export default {
  async initFilterLists(context, apolloClient) {
    if (context.state.filterListsLoading || context.state.filterLists != null) return
    await context.commit('setFilterListsLoading')

    const response = await apolloClient.query({
      query: fetchOptionListsQuery,
      fetchPolicy: 'cache-first',
    })

    // set annotationIds default according to values passed from snapshot to the store state
    context.commit('setFilterLists', {
      ...response.data,
      annotationIds: computed(() => store.state.snapshotAnnotationIds),
    })

    // Refresh the current filter so that computed defaults that depend on `filterLists` are applied
    const filter = decodeParams(context.state.route, context.state.filterLists)

    context.commit('updateFilter', filter)
  },
}
