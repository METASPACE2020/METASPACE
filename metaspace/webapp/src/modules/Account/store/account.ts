import { Module } from 'vuex';
import { DialogType } from '../dialogs';
import router from '../../../router';

export interface AccountState {
  dialog: DialogType | null;
  showDialogAsPage: boolean;
}

const account = {
  namespaced: true,
  state: {
    dialog: null,
    showDialogAsPage: false
  },
  mutations: {
    showDialog(state: AccountState, dialog: DialogType) {
      state.dialog = dialog;
      state.showDialogAsPage = false;
    },
    showDialogAsPage(state: AccountState, dialog: DialogType) {
      state.dialog = dialog;
      state.showDialogAsPage = true;
    },
    hideDialog(state: AccountState, dialog: DialogType) {
      if (state.dialog === dialog) {
        state.dialog = null;
        if (state.showDialogAsPage) {
          router.push('/');
        }
      }
    }
  }
};

export default account as Module<AccountState, any>;
