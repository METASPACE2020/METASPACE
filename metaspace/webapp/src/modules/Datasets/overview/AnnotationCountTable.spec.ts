import { mount } from '@vue/test-utils'
import router from '../../../router'
import store from '../../../store/index'
import Vue from 'vue'
import Vuex from 'vuex'
import { sync } from 'vuex-router-sync'
import { AnnotationCountTable } from './AnnotationCountTable'

Vue.use(Vuex)
sync(store, router)

describe('AnnotationCountTable', () => {
  const mockAnnotationCounts = [
    {
      databaseId: 1,
      dbName: 'HMDB',
      dbVersion: 'v4',
      counts: [
        {
          level: 5,
          n: 55,
        },
        {
          level: 10,
          n: 104,
        },
        {
          level: 20,
          n: 196,
        },
        {
          level: 50,
          n: 224,
        },
      ],
    },
    {
      databaseId: 6,
      dbName: 'ChEBI',
      dbVersion: '2018-01',
      counts: [
        {
          level: 5,
          n: 65,
        },
        {
          level: 10,
          n: 90,
        },
        {
          level: 20,
          n: 177,
        },
        {
          level: 50,
          n: 190,
        },
      ],
    },
  ]
  const mockFdrLevels = [5, 10, 20, 50]
  const propsData = { data: mockAnnotationCounts, id: 'xxxx', header: mockFdrLevels }

  const testHarness = Vue.extend({
    components: {
      AnnotationCountTable,
    },
    render(h) {
      return h(AnnotationCountTable, { props: this.$attrs })
    },
  })

  it('should match snapshot', async() => {
    const wrapper = mount(testHarness, { store, router, propsData })
    await Vue.nextTick()

    expect(wrapper).toMatchSnapshot()
  })

  it('should render the correct number of columns', async() => {
    const wrapper = mount(testHarness, { store, router, propsData })
    await Vue.nextTick()
    expect(wrapper.findAll('th').length).toBe(mockFdrLevels.length + 1)
  })

  it('should render the correct link to navigate to annotation', async() => {
    const wrapper = mount(testHarness, { store, router, propsData })
    await Vue.nextTick()

    // check if each link cell has the correct parameters
    wrapper.findAll('a').wrappers.forEach((link, index) => {
      const mockIndex = Math.floor(index / (mockFdrLevels.length || 1))
      const countIndex = index % mockFdrLevels.length
      expect(link.props('to').params.datasetId).toBe('xxxx')
      expect(link.props('to').query.db_id).toBe(mockAnnotationCounts[mockIndex].databaseId.toString())
      expect(link.props('to').query.fdr)
        .toBe((mockAnnotationCounts[mockIndex].counts[countIndex].level / 100).toString())
    })

    expect(wrapper.findAll('th').length).toBe(mockFdrLevels.length + 1)
    expect(wrapper.findAll('a').at(1).props('to').params.datasetId)
      .toBe('xxxx')
  })
})
