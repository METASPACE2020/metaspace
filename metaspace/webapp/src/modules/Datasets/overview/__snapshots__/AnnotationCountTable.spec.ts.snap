// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AnnotationCountTable > it should match snapshot 1`] = `
"<div class=\\"relative text-center\\">
  <h4># of annotations with FDR% &lt;=</h4>
  <div class=\\"el-table--fit el-table--enable-row-hover el-table--enable-row-transition has-footer el-table el-table--layout-fixed\\" style=\\"width: 100%;\\" data-prefix=\\"el\\">
    <div class=\\"el-table__inner-wrapper\\">
      <div class=\\"hidden-columns\\">
        <div></div>
        <div></div>
        <div></div>
        <div></div>
        <div></div>
      </div>
      <div class=\\"el-table__header-wrapper\\">
        <table class=\\"el-table__header\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\">
          <colgroup>
            <col name=\\"el-table_1_column_1\\" width=\\"80\\">
            <col name=\\"el-table_1_column_2\\" width=\\"80\\">
            <col name=\\"el-table_1_column_3\\" width=\\"80\\">
            <col name=\\"el-table_1_column_4\\" width=\\"80\\">
            <col name=\\"el-table_1_column_5\\" width=\\"80\\">
          </colgroup>
          <thead class=\\"\\">
            <tr class=\\"\\">
              <th class=\\"el-table_1_column_1 is-leaf is-sortable el-table__cell\\" colspan=\\"1\\" rowspan=\\"1\\">
                <div class=\\"cell\\">Database<span class=\\"caret-wrapper\\"><i class=\\"sort-caret ascending\\"></i><i class=\\"sort-caret descending\\"></i></span>
                  <!---->
                </div>
              </th>
              <th class=\\"el-table_1_column_2 is-leaf is-sortable el-table__cell\\" colspan=\\"1\\" rowspan=\\"1\\">
                <div class=\\"cell\\">5%<span class=\\"caret-wrapper\\"><i class=\\"sort-caret ascending\\"></i><i class=\\"sort-caret descending\\"></i></span>
                  <!---->
                </div>
              </th>
              <th class=\\"el-table_1_column_3 is-leaf is-sortable el-table__cell\\" colspan=\\"1\\" rowspan=\\"1\\">
                <div class=\\"cell\\">10%<span class=\\"caret-wrapper\\"><i class=\\"sort-caret ascending\\"></i><i class=\\"sort-caret descending\\"></i></span>
                  <!---->
                </div>
              </th>
              <th class=\\"el-table_1_column_4 is-leaf is-sortable el-table__cell\\" colspan=\\"1\\" rowspan=\\"1\\">
                <div class=\\"cell\\">20%<span class=\\"caret-wrapper\\"><i class=\\"sort-caret ascending\\"></i><i class=\\"sort-caret descending\\"></i></span>
                  <!---->
                </div>
              </th>
              <th class=\\"el-table_1_column_5 is-leaf is-sortable el-table__cell\\" colspan=\\"1\\" rowspan=\\"1\\">
                <div class=\\"cell\\">50%<span class=\\"caret-wrapper\\"><i class=\\"sort-caret ascending\\"></i><i class=\\"sort-caret descending\\"></i></span>
                  <!---->
                </div>
              </th>
            </tr>
          </thead>
        </table>
      </div>
      <div class=\\"el-table__body-wrapper\\">
        <div class=\\"el-scrollbar\\">
          <div class=\\"el-scrollbar__wrap el-scrollbar__wrap--hidden-default\\">
            <div class=\\"el-scrollbar__view\\" style=\\"display: inline-block; vertical-align: middle;\\">
              <table class=\\"el-table__body\\" cellspacing=\\"0\\" cellpadding=\\"0\\" border=\\"0\\" style=\\"table-layout: fixed;\\">
                <colgroup>
                  <col name=\\"el-table_1_column_1\\" width=\\"80\\">
                  <col name=\\"el-table_1_column_2\\" width=\\"80\\">
                  <col name=\\"el-table_1_column_3\\" width=\\"80\\">
                  <col name=\\"el-table_1_column_4\\" width=\\"80\\">
                  <col name=\\"el-table_1_column_5\\" width=\\"80\\">
                </colgroup>
                <!--v-if-->
                <tbody tabindex=\\"-1\\">
                  <tr class=\\"el-table__row\\">
                    <td class=\\"el-table_1_column_1 el-table__cell\\" rowspan=\\"1\\" colspan=\\"1\\">
                      <div class=\\"cell\\">
                        <!---->
                        <div class=\\"mock-router-link\\" to=\\"{&quot;name&quot;:&quot;dataset-annotations&quot;,&quot;params&quot;:{&quot;dataset_id&quot;:&quot;xxxx&quot;},&quot;query&quot;:{&quot;db_id&quot;:&quot;1&quot;,&quot;ds&quot;:&quot;xxxx&quot;}}\\">HMDB-v4</div>
                      </div>
                    </td>
                    <td class=\\"el-table_1_column_2 el-table__cell\\" rowspan=\\"1\\" colspan=\\"1\\">
                      <div class=\\"cell\\">
                        <!---->
                        <div class=\\"mock-router-link\\" to=\\"{&quot;name&quot;:&quot;dataset-annotations&quot;,&quot;params&quot;:{&quot;dataset_id&quot;:&quot;xxxx&quot;},&quot;query&quot;:{&quot;db_id&quot;:&quot;1&quot;,&quot;ds&quot;:&quot;xxxx&quot;,&quot;fdr&quot;:&quot;0.05&quot;}}\\">55</div>
                      </div>
                    </td>
                    <td class=\\"el-table_1_column_3 el-table__cell\\" rowspan=\\"1\\" colspan=\\"1\\">
                      <div class=\\"cell\\">
                        <!---->
                        <div class=\\"mock-router-link\\" to=\\"{&quot;name&quot;:&quot;dataset-annotations&quot;,&quot;params&quot;:{&quot;dataset_id&quot;:&quot;xxxx&quot;},&quot;query&quot;:{&quot;db_id&quot;:&quot;1&quot;,&quot;ds&quot;:&quot;xxxx&quot;,&quot;fdr&quot;:&quot;0.1&quot;}}\\">104</div>
                      </div>
                    </td>
                    <td class=\\"el-table_1_column_4 el-table__cell\\" rowspan=\\"1\\" colspan=\\"1\\">
                      <div class=\\"cell\\">
                        <!---->
                        <div class=\\"mock-router-link\\" to=\\"{&quot;name&quot;:&quot;dataset-annotations&quot;,&quot;params&quot;:{&quot;dataset_id&quot;:&quot;xxxx&quot;},&quot;query&quot;:{&quot;db_id&quot;:&quot;1&quot;,&quot;ds&quot;:&quot;xxxx&quot;,&quot;fdr&quot;:&quot;0.2&quot;}}\\">196</div>
                      </div>
                    </td>
                    <td class=\\"el-table_1_column_5 el-table__cell\\" rowspan=\\"1\\" colspan=\\"1\\">
                      <div class=\\"cell\\">
                        <!---->
                        <div class=\\"mock-router-link\\" to=\\"{&quot;name&quot;:&quot;dataset-annotations&quot;,&quot;params&quot;:{&quot;dataset_id&quot;:&quot;xxxx&quot;},&quot;query&quot;:{&quot;db_id&quot;:&quot;1&quot;,&quot;ds&quot;:&quot;xxxx&quot;,&quot;fdr&quot;:&quot;0.5&quot;}}\\">224</div>
                      </div>
                    </td>
                  </tr>
                  <tr class=\\"el-table__row\\">
                    <td class=\\"el-table_1_column_1 el-table__cell\\" rowspan=\\"1\\" colspan=\\"1\\">
                      <div class=\\"cell\\">
                        <!---->
                        <div class=\\"mock-router-link\\" to=\\"{&quot;name&quot;:&quot;dataset-annotations&quot;,&quot;params&quot;:{&quot;dataset_id&quot;:&quot;xxxx&quot;},&quot;query&quot;:{&quot;db_id&quot;:&quot;6&quot;,&quot;ds&quot;:&quot;xxxx&quot;}}\\">ChEBI-2018-01</div>
                      </div>
                    </td>
                    <td class=\\"el-table_1_column_2 el-table__cell\\" rowspan=\\"1\\" colspan=\\"1\\">
                      <div class=\\"cell\\">
                        <!---->
                        <div class=\\"mock-router-link\\" to=\\"{&quot;name&quot;:&quot;dataset-annotations&quot;,&quot;params&quot;:{&quot;dataset_id&quot;:&quot;xxxx&quot;},&quot;query&quot;:{&quot;db_id&quot;:&quot;6&quot;,&quot;ds&quot;:&quot;xxxx&quot;,&quot;fdr&quot;:&quot;0.05&quot;}}\\">65</div>
                      </div>
                    </td>
                    <td class=\\"el-table_1_column_3 el-table__cell\\" rowspan=\\"1\\" colspan=\\"1\\">
                      <div class=\\"cell\\">
                        <!---->
                        <div class=\\"mock-router-link\\" to=\\"{&quot;name&quot;:&quot;dataset-annotations&quot;,&quot;params&quot;:{&quot;dataset_id&quot;:&quot;xxxx&quot;},&quot;query&quot;:{&quot;db_id&quot;:&quot;6&quot;,&quot;ds&quot;:&quot;xxxx&quot;,&quot;fdr&quot;:&quot;0.1&quot;}}\\">90</div>
                      </div>
                    </td>
                    <td class=\\"el-table_1_column_4 el-table__cell\\" rowspan=\\"1\\" colspan=\\"1\\">
                      <div class=\\"cell\\">
                        <!---->
                        <div class=\\"mock-router-link\\" to=\\"{&quot;name&quot;:&quot;dataset-annotations&quot;,&quot;params&quot;:{&quot;dataset_id&quot;:&quot;xxxx&quot;},&quot;query&quot;:{&quot;db_id&quot;:&quot;6&quot;,&quot;ds&quot;:&quot;xxxx&quot;,&quot;fdr&quot;:&quot;0.2&quot;}}\\">177</div>
                      </div>
                    </td>
                    <td class=\\"el-table_1_column_5 el-table__cell\\" rowspan=\\"1\\" colspan=\\"1\\">
                      <div class=\\"cell\\">
                        <!---->
                        <div class=\\"mock-router-link\\" to=\\"{&quot;name&quot;:&quot;dataset-annotations&quot;,&quot;params&quot;:{&quot;dataset_id&quot;:&quot;xxxx&quot;},&quot;query&quot;:{&quot;db_id&quot;:&quot;6&quot;,&quot;ds&quot;:&quot;xxxx&quot;,&quot;fdr&quot;:&quot;0.5&quot;}}\\">190</div>
                      </div>
                    </td>
                  </tr>
                </tbody>
                <!--v-if-->
              </table>
              <!--v-if-->
              <!--v-if-->
            </div>
          </div>
          <div class=\\"el-scrollbar__bar is-horizontal\\" style=\\"display: none;\\">
            <div class=\\"el-scrollbar__thumb\\" style=\\"transform: translateX(0%);\\"></div>
          </div>
          <div class=\\"el-scrollbar__bar is-vertical\\" style=\\"display: none;\\">
            <div class=\\"el-scrollbar__thumb\\" style=\\"transform: translateY(0%);\\"></div>
          </div>
        </div>
      </div>
      <div class=\\"el-table__footer-wrapper\\">
        <table class=\\"el-table__footer\\" cellspacing=\\"0\\" cellpadding=\\"0\\" border=\\"0\\">
          <colgroup>
            <col name=\\"el-table_1_column_1\\" width=\\"80\\">
            <col name=\\"el-table_1_column_2\\" width=\\"80\\">
            <col name=\\"el-table_1_column_3\\" width=\\"80\\">
            <col name=\\"el-table_1_column_4\\" width=\\"80\\">
            <col name=\\"el-table_1_column_5\\" width=\\"80\\">
          </colgroup>
          <tfoot>
            <tr>
              <td colspan=\\"1\\" rowspan=\\"1\\" class=\\"el-table__cell el-table_1_column_1 is-leaf\\">
                <div class=\\"cell\\">Total Annotations</div>
              </td>
              <td colspan=\\"1\\" rowspan=\\"1\\" class=\\"el-table__cell el-table_1_column_2 is-leaf\\">
                <div class=\\"cell\\">
                  <div class=\\"mock-router-link\\" to=\\"{&quot;name&quot;:&quot;dataset-annotations&quot;,&quot;params&quot;:{&quot;dataset_id&quot;:&quot;xxxx&quot;},&quot;query&quot;:{&quot;ds&quot;:&quot;xxxx&quot;,&quot;fdr&quot;:&quot;0.05&quot;}}\\">120</div>
                </div>
              </td>
              <td colspan=\\"1\\" rowspan=\\"1\\" class=\\"el-table__cell el-table_1_column_3 is-leaf\\">
                <div class=\\"cell\\">
                  <div class=\\"mock-router-link\\" to=\\"{&quot;name&quot;:&quot;dataset-annotations&quot;,&quot;params&quot;:{&quot;dataset_id&quot;:&quot;xxxx&quot;},&quot;query&quot;:{&quot;ds&quot;:&quot;xxxx&quot;,&quot;fdr&quot;:&quot;0.1&quot;}}\\">194</div>
                </div>
              </td>
              <td colspan=\\"1\\" rowspan=\\"1\\" class=\\"el-table__cell el-table_1_column_4 is-leaf\\">
                <div class=\\"cell\\">
                  <div class=\\"mock-router-link\\" to=\\"{&quot;name&quot;:&quot;dataset-annotations&quot;,&quot;params&quot;:{&quot;dataset_id&quot;:&quot;xxxx&quot;},&quot;query&quot;:{&quot;ds&quot;:&quot;xxxx&quot;,&quot;fdr&quot;:&quot;0.2&quot;}}\\">373</div>
                </div>
              </td>
              <td colspan=\\"1\\" rowspan=\\"1\\" class=\\"el-table__cell el-table_1_column_5 is-leaf\\">
                <div class=\\"cell\\">
                  <div class=\\"mock-router-link\\" to=\\"{&quot;name&quot;:&quot;dataset-annotations&quot;,&quot;params&quot;:{&quot;dataset_id&quot;:&quot;xxxx&quot;},&quot;query&quot;:{&quot;ds&quot;:&quot;xxxx&quot;,&quot;fdr&quot;:&quot;0.5&quot;}}\\">414</div>
                </div>
              </td>
            </tr>
          </tfoot>
        </table>
      </div>
      <!--v-if-->
    </div>
    <div class=\\"el-table__column-resize-proxy\\" style=\\"display: none;\\"></div>
  </div>
  <div class=\\"text-right mt-2\\"><button ariadisabled=\\"false\\" type=\\"button\\" class=\\"el-button\\">
      <!--v-if--><span class=\\"\\">Browse annotations</span>
    </button></div>
</div>"
`;
